wpilibNewCommands/src/main/java/edu/wpi/first/wpilibj2/command/CommandScheduler.java

```diff
@@ -84,9 +86,11 @@ public static synchronized CommandScheduler getInstance() {
 
   // Flag and queues for avoiding ConcurrentModificationException if commands are
   // scheduled/canceled during run
-  private boolean m_inRunLoop;
-  private final Set<Command> m_toSchedule = new LinkedHashSet<>();
-  private final List<Command> m_toCancel = new ArrayList<>();
+  private boolean m_deferCancel;
+  private boolean m_deferSchedule;
```

Comment:
This should be a shared flag; we don't want to defer only one.

Reply:
It might need to be split; if a command is scheduled during `end`, the schedule gets deferred to the next loop, which is a behavior change.

Reply:
Both operations should be queued when entering the "sensitive" sections, and polled when exiting that section.

Reply:
Not applicable anymore; PR's been rewritten to use sets based on @KangarooKoala's feedback.